src/node/txpackagetracker.cpp

```diff
@@ -0,0 +1,181 @@
+        for (const auto& wtxid : block_wtxids) {
+            orphan_request_tracker.ForgetTxHash(wtxid);
+        }
+        for (const auto& wtxid : conflicted_wtxids) {
+            orphan_request_tracker.ForgetTxHash(wtxid);
+        }
+    }
+    bool HaveTxToReconsider(NodeId peer) { return m_orphanage.HaveTxToReconsider(peer); }
+    size_t OrphanageSize() { return m_orphanage.Size(); }
+    void MempoolAcceptedTx(const CTransactionRef& ptx) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        LOCK(m_mutex);
+        m_orphanage.AddChildrenToWorkSet(*ptx);
+        m_orphanage.EraseTx(ptx->GetWitnessHash());
+    }
+    void MempoolRejectedTx(const uint256& wtxid) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        LOCK(m_mutex);
+        m_orphanage.EraseTx(wtxid);
+    }
+    void AddOrphanTx(NodeId nodeid, const uint256& wtxid, const CTransactionRef& tx, bool is_preferred, std::chrono::microseconds reqtime)
+        EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        AssertLockNotHeld(m_mutex);
+        LOCK(m_mutex);
+        // Skip if we weren't provided the tx and can't find the wtxid in the orphanage.
+        if (tx == nullptr && !m_orphanage.HaveTx(GenTxid::Wtxid(wtxid))) return;
+
+        // Even though this stores the orphan wtxid, GenTxid::Txid instead of Wtxid because we will be requesting the parents via txid.
+        orphan_request_tracker.ReceivedInv(nodeid, GenTxid::Txid(wtxid), is_preferred, reqtime);
+
+        if (tx != nullptr) {
+            m_orphanage.AddTx(tx, nodeid);
+        } else {
+            m_orphanage.AddTx(m_orphanage.GetTx(wtxid), nodeid);
+        }
+        // DoS prevention: do not allow m_orphanage to grow unbounded (see CVE-2012-3789)
+        m_orphanage.LimitOrphans(m_max_orphanage_count);
+    }
+    size_t CountInFlight(NodeId nodeid) const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        AssertLockNotHeld(m_mutex);
+        LOCK(m_mutex);
+        auto count{orphan_request_tracker.CountInFlight(nodeid)};
+        return count;
+    }
+    size_t Count(NodeId nodeid) const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        AssertLockNotHeld(m_mutex);
+        LOCK(m_mutex);
+        auto count{orphan_request_tracker.Count(nodeid)};
+        return count;
+    }
+
+    std::vector<GenTxid> GetOrphanRequests(NodeId nodeid, std::chrono::microseconds current_time)
+        EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        AssertLockNotHeld(m_mutex);
+        LOCK(m_mutex);
+        std::vector<std::pair<NodeId, GenTxid>> expired;
+        auto tracker_requestable = orphan_request_tracker.GetRequestable(nodeid, current_time, &expired);
+        for (const auto& entry : expired) {
+            LogPrint(BCLog::TXPACKAGES, "\nTimeout of inflight %s %s from peer=%d\n", entry.second.IsWtxid() ? "ancpkginfo" : "orphan parent",
+                entry.second.GetHash().ToString(), entry.first);
+        }
+        std::vector<GenTxid> results;
+        for (const auto& gtxid : tracker_requestable) {
```

Comment:
Would you mind adding some spacing around logical blocks in your code? I find a lot of the new code hard to read with a lack of spacing

e.g.:
```suggestion
    {
        AssertLockNotHeld(m_mutex);
        LOCK(m_mutex);

        std::vector<std::pair<NodeId, GenTxid>> expired;
        auto tracker_requestable = orphan_request_tracker.GetRequestable(nodeid, current_time, &expired);
        for (const auto& entry : expired) {
            LogPrint(BCLog::TXPACKAGES, "\nTimeout of inflight %s %s from peer=%d\n", entry.second.IsWtxid() ? "ancpkginfo" : "orphan parent",
                entry.second.GetHash().ToString(), entry.first);
        }

        std::vector<GenTxid> results;
        for (const auto& gtxid : tracker_requestable) {
```